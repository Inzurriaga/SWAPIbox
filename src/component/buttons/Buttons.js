import React, { Component } from "react"

export default class Buttons extends Component{
    switchContainer = (container) => {
        this.props.switchContainer(container)
    }

    render(){
        return(
            <nav>
                <div>
                    <button data-text="People" onClick={() => {this.switchContainer("peopleContainer")}}>Peoples</button>
                </div>
                <div>
                    <button data-text="Planets" onClick={() => {this.switchContainer("planetsContainer")}}>Planets</button>
                </div>
                <div>
                    <button data-text="Vehicles" 
                onClick={() => {this.switchContainer("vehiclesContainer")}}>Vehicles</button>
                </div>
            </nav>
        )
    }
}