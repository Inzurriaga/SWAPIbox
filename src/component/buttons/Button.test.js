import React from "react"
import { shallow } from "enzyme"
import Buttons from "./Buttons"

const mockSwitchContainer = jest.fn()

describe("button", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <Buttons switchContainer={mockSwitchContainer}/>
        )
    })
    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()
    })
    it("should invoke the the mock function", () => {
        wrapper.find("button").at(0).simulate("click")
        expect(mockSwitchContainer).toHaveBeenCalled()
    })
})