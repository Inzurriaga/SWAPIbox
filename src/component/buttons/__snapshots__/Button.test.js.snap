// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot 1`] = `
<nav>
  <div>
    <button
      data-text="People"
      onClick={[Function]}
    >
      Peoples
    </button>
  </div>
  <div>
    <button
      data-text="Planets"
      onClick={[Function]}
    >
      Planets
    </button>
  </div>
  <div>
    <button
      data-text="Vehicles"
      onClick={[Function]}
    >
      Vehicles
    </button>
  </div>
</nav>
`;
